src/pages/en/core-concepts/layouts.md

```diff
@@ -9,103 +9,263 @@ i18nReady: true
 
 A layout component is conventionally used to provide an [`.astro` or `.md` page](/en/core-concepts/astro-pages/) both a **page shell** (`<html>`, `<head>` and `<body>` tags) and a `<slot />` to specify where in the layout page content should be injected.
 
-Layouts often provide common `<head>` elements and common UI elements for the page such as headers, navigation bars and footers.
+Layouts, like any Astro component, can also [accept props](/en/core-concepts/astro-components/#component-props) and [import and use other components](/en/core-concepts/astro-components/#component-structure). This can be useful for for common `<head>` elements and common UI elements for the page such as headers, navigation bars and footers.
 
 Layout components are commonly placed in a `src/layouts` directory in your project.
 
 ## Sample Layout
 
 **`src/layouts/MySiteLayout.astro`**
 
-```astro
+```astro "<slot />" 
 ---
+import BaseHead from '../components/BaseHead.astro';
+import Footer from '../components/Footer.astro';
+const { title } = Astro.props
 ---
 <html lang="en">
   <head>
     <meta charset="utf-8">
-    <title>My Cool Astro Site</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
+    <BaseHead title={title}/>
   </head>
   <body>
     <nav>
       <a href="#">Home</a>
       <a href="#">Posts</a>
       <a href="#">Contact</a>
     </nav>
+    <h1>{title}</h1>
     <article>
       <slot /> <!-- your content is injected here -->
     </article>
+    <Footer />
   </body>
 </html>
 ```
 
-**`src/pages/index.astro`**
+**`src/pages/index.astro`** 
 
-```astro {2} /</?MySiteLayout>/
+```astro {2} /</MySiteLayout>/  /<MySiteLayout title=/ '"Home Page">'
 ---
 import MySiteLayout from '../layouts/MySiteLayout.astro';
 ---
-<MySiteLayout>
+<MySiteLayout title="Home Page">
   <p>My page content, wrapped in a layout!</p>
 </MySiteLayout>
 ```
 
 
 📚 Learn more about [slots](/en/core-concepts/astro-components/#slots).
 
-## Markdown Layouts
+## Markdown/MDX Layouts
 
-Page layouts are especially useful for [Markdown files](/en/guides/markdown-content/#markdown-and-mdx-pages). Markdown files can use a special `layout` property at the top of the frontmatter to specify which `.astro` component to use as a page layout.
+Page layouts are especially useful for [Markdown and MDX files](/en/guides/markdown-content/#markdown-and-mdx-pages) which otherwise would not have any page formatting. 
```

Comment:
Nit: but “pages” over “files” because those specifically can get turned into a route directly.

```suggestion
Page layouts are especially useful for [Markdown and MDX pages](/en/guides/markdown-content/#markdown-and-mdx-pages) which otherwise would not have any page formatting. 
```

(Not doing well on keeping zoomed out…)